src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/TagHelperFeature.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// The list of tag helper types in an MVC application. The <see cref="TagHelperFeature"/> can be populated
+    /// using the <see cref="ApplicationPartManager"/> that is available during startup at <see cref="IMvcBuilder.PartManager"/>
+    /// and <see cref="IMvcCoreBuilder.PartManager"/> or at a later stage by requiring the <see cref="ApplicationPartManager"/>
+    /// as a dependency in a component.
+    /// </summary>
+    public class TagHelperFeature
+    {
+        public IList<TypeInfo> TagHelpers { get; } = new List<TypeInfo>();
```

Comment:
Needs docs. Also, I see all over you're doing `AsType`. Would it be simpler to just have this be an `IList<Type>`?